modules/backend/behaviors/ListController.php

```diff
@@ -362,6 +362,103 @@ public function index_onDelete()
         return $this->controller->listRefresh($definition);
     }
 
+    /**
+     * Bulk copy records.
+     * @return void
+     * @throws \Winter\Storm\Exception\ApplicationException when the parent definition is missing.
+     */
+    public function index_onCopy()
+    {
+        if (method_exists($this->controller, 'onCopy')) {
+            return call_user_func_array([$this->controller, 'onCopy'], func_get_args());
+        }
+
+        /*
+         * Establish the list definition
+         */
+        $definition = post('definition', $this->primaryDefinition);
+
+        if (!isset($this->listDefinitions[$definition])) {
+            throw new ApplicationException(Lang::get('backend::lang.list.missing_parent_definition', compact('definition')));
+        }
+
+        $listConfig = $this->controller->listGetConfig($definition);
+
+        /*
+         * Validate checked identifiers
+         */
+        $checkedIds = post('checked');
+
+        if (!$checkedIds || !is_array($checkedIds) || !count($checkedIds)) {
+            Flash::error(Lang::get(
+                (!empty($listConfig->noRecordsCopiedMessage))
+                    ? $listConfig->noRecordsCopiedMessage
+                    : 'backend::lang.list.copy_selected_empty'
+            ));
+            return $this->controller->listRefresh();
+        }
+
+        /*
+         * Create the model
+         */
+        $class = $listConfig->modelClass;
+        $model = new $class;
+        $model = $this->controller->listExtendModel($model, $definition);
+
+        /*
+         * Create the query
+         */
+        $query = $model->newQuery();
+        $this->controller->listExtendQueryBefore($query, $definition);
+
+        $query->whereIn($model->getKeyName(), $checkedIds);
+        $this->controller->listExtendQuery($query, $definition);
+
+        /*
+         *  Retrieve protected columns from list column configuration
+         */
+        $columnConfig = $this->makeConfig($listConfig->list);
+
+        $protectedColumns = [];
+
+        foreach ($columnConfig as $column => $definition) {
+            if (array_key_exists('protected', $definition)) {
+                if ($definition['protected']) {
+                    $protectedColumns[] = $column;
+                }
+            }
+        }
+
+        /*
+         * Copy records
+         */
+        $records = $query->get();
+
+        if ($records->count()) {
+            foreach ($records as $record) {
+                $copy = $record->replicate($protectedColumns);
+                $this->controller->listBeforeCopy($record);
+                $copy->save();
+                $this->controller->listAfterCopy($record);
```

Comment:
There might be a need to have some error checking or a try / catch block here, in case a record cannot be copied due to validation rules or uniqueness checks?